lib/spack/spack/tag.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+"""Classes and functions to manage package tags"""
+import collections
+import copy
+import sys
+
+if sys.version_info >= (3, 5):
+    from collections.abc import Mapping  # novm
+else:
+    from collections import Mapping
+
+import spack.error
+import spack.util.spack_json as sjson
+
+
+def get_installed_packages():
+    """Returns names of packages installed in the active environment."""
+    env = spack.environment.active_environment()
+    hashes = env.all_hashes() if env else None
+    return [spec.name for spec in spack.store.db.query(hashes=hashes)]
+
+
+def get_tag_packages(tags, installed, skip_empty):
```

Comment:
What do you think of:
```suggestion
def packages_with_tags(tags, installed, skip_empty):
```
as a name for the function?